doc/man3/OSSL_SELF_TEST_new.pod

```diff
@@ -0,0 +1,177 @@
+=pod
+
+=head1 NAME
+
+OSSL_SELF_TEST_new,
+OSSL_SELF_TEST_free,
+OSSL_SELF_TEST_onbegin,
+OSSL_SELF_TEST_oncorrupt_byte,
+OSSL_SELF_TEST_onend - functionality to trigger a callback during a self test
+
+=head1 SYNOPSIS
+
+ #include <openssl/self_test.h>
+
+ OSSL_SELF_TEST *OSSL_SELF_TEST_new(OSSL_CALLBACK *cb, void *cbarg);
+ void OSSL_SELF_TEST_free(OSSL_SELF_TEST *st);
+
+ void OSSL_SELF_TEST_onbegin(OSSL_SELF_TEST *st, const char *type,
+                             const char *desc);
+ void OSSL_SELF_TEST_oncorrupt_byte(OSSL_SELF_TEST *st, unsigned char *bytes);
+ void OSSL_SELF_TEST_onend(OSSL_SELF_TEST *st, int ret);
+
+=head1 DESCRIPTION
+
+These methods are intended for use by provider implementors, to display
+diagnostic information during self testing.
+
+There are 3 types of self tests that can be set up:
+
+=over 4
+
+=item "Power on Self Tests" (POST)
+
+During startup a provider may want to run known answer tests on algorithms to
+test that the system is operating correctly.
+
+=item "On demand Self tests"
+
+The application may manually trigger the POST via an API call.
+
+=item "Conditional tests"
+
+Used during an operation on an algorithm e.g. when performing a pairwise test
+during key generation of asymmetric keys.
+
+=back
+
+OSSL_SELF_TEST_new() allocates an opaque B<OSSL_SELF_TEST> object that has a
+callback and callback argument associated with it. The callback is triggered by
+the 3 phases of the self test that can be setup.
+
+The following OSSL_PARAMS are passed to the callback during each phase:
+
+=over 4
+
+=item "st-phase" (B<OSSL_PROV_PARAM_SELF_TEST_PHASE>) <UTF8 string>
+
+The value is one of:
+
+=over 4
+
+=item "Start" (B<OSSL_SELF_TEST_PHASE_START>) <UTF8 string>
+
+This is sent by OSSL_SELF_TEST_onbegin().
+
+=item "Corrupt" (B<OSSL_SELF_TEST_PHASE_CORRUPT>) <UTF8 string>
+
+This is sent by OSSL_SELF_TEST_oncorrupt_byte().
+
+=item "Pass" (B<OSSL_SELF_TEST_PHASE_PASS>) <UTF8 string>
+
+=item "Fail" (B<OSSL_SELF_TEST_PHASE_FAIL>) <UTF8 string>
+
+The value "Fail" is sent by OSSL_SELF_TEST_onend() if the B<ret> value is 0,
+otherwise it sends the value "Pass".
+
+=back
+
+=item "st-type" (B<OSSL_PROV_PARAM_SELF_TEST_TYPE>) <UTF8 string>
+
+The value is setup by the B<type> passed to OSSL_SELF_TEST_onbegin().
```

Comment:
Use `I` for variables, i.e. `I<type>` in this case.  Bold is not conformant to man-page standards.